### ISSUE TITLE
Is it possibile to controll the number of seconds the input audio file is chuncked?

### ISSUE THREAD

**Issue by Lorenzoncina** [guillaumekln/faster-whisper#428]
Input audio file is chucked every 30 seconds and then transcribed. 
Hugging face API allows, instead, to controll this value while loading the whisper model through the pipeline,  check this for reference:https://huggingface.co/blog/asr-chunking

Is there a way in faster-whisper to reduce these 30 seconds to a lower value?

**Comment by guillaumekln** [comment]
> Input audio file is chucked every 30 seconds and then transcribed.

That's not entirely correct. The Whisper algorithm uses a 30-second *sliding* window, and the window is advanced depending on the model output.

Note that the Whisper model was trained with 30-second inputs. You typically can't reduce this size, even if Hugging Face makes it configurable. See also #171.

**Comment by Lorenzoncina** [comment]
Thank you very much for the answer and reference Guillaume!
The point here it's, I've fine tuned Whisper large V2 (as well as other models) and, once converted to faster-whisper, they perform poorly on long audio files (with respect to pre-trained one), in particular they have deletion errors.  Instead, on Hugging Face, with chunck_lenght = 10 seconds, there are way less deletion erorrs. There could be thounds of reasons, but one could be that I've used train data with shorte audio file (around 10 seconds). 

Do you think there could a correletion?

**Comment by YaCodesDevelopment** [comment]
Seeing exactly the same behavior with fine-tuned whisper medium model (has nearly ideal performance on short audio test set, but using with long audio in faster-whisper results into some gibberish crap, especially in pieces with a lot of silence.

**Comment by Lorenzoncina** [comment]
@YaCodesDevelopment you can try inference with Hugging face api on your fine-tuned model, specifing chunck_lenght = 10 seconds anche check if you get a better output. In my case it improves the output since I fine tuned with shorter audio file (around 10-15 seconds). Let me know!

THe problem of Hugginf face api is that they don't have a VAD to filter out silence like faster whisper, morover they are slow and consume a lot of resources so I'm not considering to use them. Instead I'm trying to modify this chunkc lenght here on faster-whisper but without any secces yet

**Comment by YaCodesDevelopment** [comment]
I'll try, the only thing that I don't have my model at HuggingFace, but I'll see how to deal with that. Generally, after switching to whisper.cpp + cuBLAS I got x5 times longer inference time (on faster whisper I've used INT8 and GPU to run it entirely, so that might me the thing), but with that quality did also drastically increase + I've stopped getting CUDA OOM, as VAD didn't work for me there for some reason, would always throw some error I didn't have time to debug, resulting into some weird behavior on my 2GB GPU.

**Comment by asusdisciple** [comment]
Had the same problem, try setting `condition_on_previous_text=False`

**Comment by thiswillbeyourgithub** [comment]
Btw I made a [comment that might be of interest here](https://github.com/SYSTRAN/faster-whisper/issues/171#issuecomment-2389369891) :

> Hi, for your information [FUTO.org](https://www.futo.org/about/what-is-futo/) did a finetuning of whisper that can work with [arbitrarily less audio than 30s](https://github.com/futo-org/whisper-acft) and made it handy to use on smartphones via [their android keyboard app](https://github.com/futo-org/android-keyboard/releases). I [don't think it is yet using quantization NOR faster-whisper NOR v3 turbo models so I opened this issue to ask if it's planned](https://github.com/futo-org/whisper-acft/issues/9).

> They put their code in a notebook on their repo so if anyone is interested in contributing I think it might be a high value contribution as the current models are already awesome on smartphones, even medium powered one! But of course they could benefit from faster inference or more compact size.

> I **think** that the main dev doing this at futo has more expertise on keyboard apps and android developpments in general than ML / sound processing so this could be a real help (although I might be wrong about his skills!)


> Related: #456 and #428

**Comment by VamsiMarriwada** [comment]
> Thank you very much for the answer and reference Guillaume! The point here it's, I've fine tuned Whisper large V2 (as well as other models) and, once converted to faster-whisper, they perform poorly on long audio files (with respect to pre-trained one), in particular they have deletion errors. Instead, on Hugging Face, with chunck_lenght = 10 seconds, there are way less deletion erorrs. There could be thounds of reasons, but one could be that I've used train sets with shorter audio files (around 10 seconds).
> 
> Do you think there could a correletion?

Hi sir, did you find any solution for this issue, I too face this issue with my fine tuned whisper medium model

**Comment by VamsiMarriwada** [comment]
> Seeing exactly the same behavior with fine-tuned whisper medium model (has nearly ideal performance on short audio test set, but using with long audio in faster-whisper results into some gibberish crap, especially in pieces with a lot of silence.

Hi sir, did you find any solution for this issue, I too face this issue with my fine tuned whisper medium model
